### DISCUSSION TITLE
error in exporttopdf button

### DISCUSSION THREAD

**Discussion by obaidalrushdi** [serenity-is/Serenity#5676] [Help]
Hi ,
I added exporttopdf button to grid but when i click on it i got this errer
Message: Uncaught ReferenceError: jsPDF is not defined

File: http://localhost:59952/Scripts/site/osystemy.Web.js?v=EdPwTHcrO4hdUqQ_2HWhEQ, Line: 5989, Column: 35
i am using serenity .net core 5

**Comment by obaidalrushdi** [comment]
i solve the problem by copy the file jspdf.js to script folder

**Reply by sooneeatwork** [reply]
hi, i also faced the same issues but my problem persists even though the jspdf.js is in script folder

**Reply by sooneeatwork** [reply]
i think after we migrate to .net 5, the pdf should refer to serenity extension js right? how come we still need to refer to script folder js

**Reply by imhemayatsangin** [reply]
> i think after we migrate to .net 5, the pdf should refer to serenity extension js right? how come we still need to refer to script folder js

Did you solve this issue? please if you can guide me im getting the same error and the jspdf.js file is in the Scripts directory.
Please need your help if you have done it.

**Reply by Bubyana74** [reply]
Dear, the same thing is happening to me as Sangincode404
They will be so kind to help us for the solution.

Thanks greetings

**Reply by imhemayatsangin** [reply]
Dear @Bubyana74 , I am using asp.net core template of serenity and I just copied the scripts files in the Script folder and then add reference of these files in the _references.js and it is working like a horse. You can go for more details to my this issue:
https://github.com/serenity-is/Serenity/discussions/6019

**Reply by wirble** [reply]
My version is 5.2.3 using core and I have same problem.  It's mentioned in the migration document where it should be added via the tsconfig file but for some reason it is not.  https://serenity.is/Docs/migration/v5-0-26-to-v5-0-41

**Reply by wirble** [reply]
Apparently this function has been moved to Serenity.Extensions and with the migration, it may not be installed by default.  It is mentioned in this issue #5826
